pdata/xpdata/pref/metrics.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pref
+
+import (
+	"reflect"
+
+	"go.opentelemetry.io/collector/pdata/internal"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// MarkPipelineOwnedMetrics marks the pmetric.Metrics data as owned by the pipeline, returns true if the data were
+// previously not owned by the pipeline, otherwise false.
+func MarkPipelineOwnedMetrics(md pmetric.Metrics) bool {
+	return internal.GetMetricsState(internal.Metrics(md)).MarkPipelineOwned()
+}
+
+func RefMetrics(md pmetric.Metrics) {
+	internal.GetMetricsState(internal.Metrics(md)).Ref()
+}
+
+func UnrefMetrics(md pmetric.Metrics) {
+	if !internal.GetMetricsState(internal.Metrics(md)).Unref() {
+		return
+	}
+	// Don't call DeleteOrigExportLogsServiceRequest without the gate because we reset the data and that may still cause issues.
+	if internal.UseProtoPooling.IsEnabled() {
```

Comment:
Should we bring both pooling and refcounting one one feature gate?

Reply:
They are, since refcounting is only needed for being able to do pooling.

Reply:
But we still do the counting even if it's not being used?

Reply:
Problems means that it crashes on that panic correct? But that is the core part of this, I can add an extra featuregate (default enabled) to protect that logic just in case. Would that help?